test/e2e/kubevirt.go

```diff
@@ -870,160 +876,402 @@ passwd:
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: namespace,
+					Name:      vmiName,
+				},
+			}
+			waitVirtualMachineInstanceReadiness(vmi)
+			Expect(crClient.Get(context.TODO(), crclient.ObjectKeyFromObject(vmi), vmi)).To(Succeed())
 
-	if err := client.List(context.Background(), unstructuredVMIMigrations); err != nil {
-		return nil, err
-	}
-	if len(unstructuredVMIMigrations.Items) == 0 {
-		return nil, fmt.Errorf("empty migration list")
-	}
+			step := by(vmi.Name, "Login to virtual machine for the first time")
+			Expect(kubevirt.LoginToFedora(vmi, "core", "fedora")).To(Succeed(), step)
+			expectedAddreses = virtualMachineAddressesFromStatus(vmi, 2 /*two addresses, dual stack*/)
 
-	var migrations []kubevirtv1.VirtualMachineInstanceMigration
-	for i := range unstructuredVMIMigrations.Items {
-		var vmiMigration kubevirtv1.VirtualMachineInstanceMigration
-		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(
-			unstructuredVMIMigrations.Items[i].Object,
-			&vmiMigration,
-		); err != nil {
-			return nil, err
-		}
-		migrations = append(migrations, vmiMigration)
-	}
+			step = by(vmi.Name, fmt.Sprintf("Check east/west traffic before %s %s", td.resource.description, td.test.description))
+			testPodsIPs := httpServerTestPodsMultusNetworkIPs(networkName)
 
-	return migrations, nil
-}
+			// kubevirt secondary IPAM do not support IPv6, so guest is not
+			// going to have an ipv6 address at the interface
+			testPodsIPs = filterOutIPv6(testPodsIPs)
+
+			checkEastWestTraffic(vmi, testPodsIPs, step)
+
+			td.test.cmd()
+			step = by(vm.Name, fmt.Sprintf("Login to virtual machine after %s %s", td.resource.description, td.test.description))
+			Expect(kubevirt.LoginToFedora(vmi, "core", "fedora")).To(Succeed(), step)
+			obtainedAddresses := virtualMachineAddressesFromStatus(vmi, 2 /*two addresses, dual stack*/)
+			Expect(obtainedAddresses).To(Equal(expectedAddreses))
+
+			step = by(vmi.Name, fmt.Sprintf("Check east/west traffic after %s %s", td.resource.description, td.test.description))
+			checkEastWestTraffic(vmi, testPodsIPs, step)
+		},
+			func(td testData) string {
+				return fmt.Sprintf("after %s of %s with %s", td.test.description, td.resource.description, td.topology)
+			},
+			Entry(nil, testData{
+				resource: virtualMachine,
+				test:     restart,
+				topology: "localnet",
+			}),
+			Entry(nil, testData{
+				resource: virtualMachine,
+				test:     restart,
+				topology: "layer2",
+			}),
+			Entry(nil, testData{
+				resource: virtualMachine,
+				test:     liveMigrate,
+				topology: "localnet",
+			}),
+			Entry(nil, testData{
+				resource: virtualMachine,
+				test:     liveMigrate,
+				topology: "layer2",
+			}),
+			Entry(nil, testData{
+				resource: virtualMachineInstance,
+				test:     liveMigrate,
+				topology: "localnet",
+			}),
+			Entry(nil, testData{
+				resource: virtualMachineInstance,
+				test:     liveMigrate,
+				topology: "layer2",
+			}),
```

Comment:
What's the point of these two? Or why don't we have restart test for VMIs? Out of curiosity.

Reply:
You cannot restart VMIs, you can only restart VMs, as far as I know we decided to also support VMIs.